lib/smoke_test.rb

```diff
@@ -0,0 +1,69 @@
+require 'net/http'
+require 'json'
+require 'openssl'
+
+class SmokeTest
+  def initialize
+    secrets         = Rails.application.secrets
+    @urls           = secrets.smoke_test[:urls]
+    @http_header    = secrets.smoke_test[:http_header]
+    @api_token      = secrets.smoke_test[:api_token] || ""
+    @slack_uri      = secrets.smoke_test[:slack_uri]
+    @slack_token    = secrets.smoke_test[:slack_token]
+    @slack_room     = secrets.smoke_test[:slack_room]
+    @slack_username = secrets.smoke_test[:slack_username]
+    @slack_emoji    = secrets.smoke_test[:slack_emoji]
+  end
+
+  def test_endpoints
+    message = ""
+
+    if @urls.nil?
+      message << "Urls are not properly configured"
+      slack_smoke_notification message
+      return
+    end
+
+    @urls.each do |url|
```

Comment:
I feel like we could break this up into some smaller methods, remember if you're writting a lot of lines of code, there's probably an easier way/it could be broken down into smaller chunks